tests/persistence/test_persistence_leverage.py

```diff
@@ -0,0 +1,638 @@
+from datetime import datetime, timedelta
+from math import isclose
+
+import pytest
+
+from freqtrade.enums import InterestMode
+from freqtrade.persistence import Trade
+from tests.conftest import log_has_re
+
+
+@pytest.mark.usefixtures("init_persistence")
+def test_interest_kraken_lev(market_lev_buy_order, fee):
+    """
+        Market trade on Kraken at 3x and 5x leverage
+        Short trade
+        interest_rate: 0.05%, 0.25% per 4 hrs
+        open_rate: 0.00004099 base
+        close_rate: 0.00004173 base
+        stake_amount: 0.0037707443218227
+        borrowed: 0.0075414886436454
+        amount:
+            275.97543219 crypto
+            459.95905365 crypto
+        borrowed:
+            0.0075414886436454  base
+            0.0150829772872908  base
+        time-periods: 10 minutes(rounds up to 1 time-period of 4hrs)
+                        5 hours = 5/4
+
+        interest: borrowed * interest_rate * ceil(1 + time-periods)
+                = 0.0075414886436454 * 0.0005  * ceil(2)   = 7.5414886436454e-06 base
+                = 0.0075414886436454 * 0.00025 * ceil(9/4) = 5.65611648273405e-06 base
+                = 0.0150829772872908 * 0.0005  * ceil(9/4) = 2.26244659309362e-05 base
+                = 0.0150829772872908 * 0.00025 * ceil(2)   = 7.5414886436454e-06 base
+    """
+
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.0037707443218227,
+        amount=275.97543219,
+        open_rate=0.00001099,
```

Comment:
I think this would be a lot simpler by using "USDT" pairs (simply because our mind calculations work better with numbers > 0).
it'll make the calculation easier (as the result will not be 1e-06).

Also - the precision used above is unrealistic - no exchange will allow you 10 decimals on any number (the maximum is usually 8 for BTC pairs, and 3-4 for USDT pairs iirc).

THis also makes "recalculating" this with the [kraken fee calculator](https://kraken-fees-calculator.github.io/) impossible - as they don't allow these numbers.

Reply:
I sent kraken an email about that calculator and [this is the coversation we had](https://i.imgur.com/i47aehi.png)

should I update the tests?

Reply:
mmmh, so relying on that might be risky (not security-wise - but to get the calculation wrong).

That's not to say it IS wrong - but it means we'll need to scrutinize the calculator before determining "these are the calculations kraken uses".

Reply:
> i find it strange that using different values for leverage (no other change) does not change the "gross margin" - only the fees (so assuming an invest of 1000, with open price 1 and closing price 1.1021 always gives 102.10 gross margin - at no leverage and at 5x leverage).

The way leverage on Kraken works is the amount is always the same, but the stake_amount is lower when you use higher leverage. So if your spending `10 USDT` to get `10 BUSD` and you apply `2x` leverage, then you're spending `5 USDT` to get `10 BUSD`, with 3x leverage, you're spending `3.3333 USDT` for `10 BUSD`

That might explain some of it, it's [explained here](https://support.kraken.com/hc/en-us/articles/203053116-How-leverage-works-in-spot-transactions-on-margin)

Reply:
I might end up dropping support for Kraken if we don't implement cross margin right away, I could possibly add support for... Bybit? Was that the other one you suggested?

Reply:
FTX ... i'm honestly not keen on bybit - as it's currently not a supported exchange (which basically means, i never ran anything on it) - and adding a new exchange as part of this is surely not a good idea.

Reply:
Oh ok, what are all the supported exchanges? I know that bittrex doesn't have margin trading at all

Reply:
Let's add FTX as another PR after this one get's merged?

Reply:
for sure - i'm not even sure the decision on "which" exchanges needs to be taken at this point ... 

I'm just ... hesitant to only add binance - as i think their API is the strangest of all.

Reply:
Seeing as the liquidation price would be the only thing which would change, I'll keep writing support for kraken, and can test it by using a stoploss high enough that the liquidation price would never trigger. I'll make sure kraken is disabled before merging onto develop also. This way in order to enabled margin on kraken, the only feature that would need to be added would be a maintenance margin